### DISCUSSION TITLE
How to add a custom module path with `postcss-import` config

### DISCUSSION THREAD

**Discussion by scottcarlson** [nuxt/nuxt#19376] [Questions]
We are porting a Nuxt 2 project to Nuxt 3, and we have custom postcss import files that we use to define postcss mixins. On Nuxt 2 when we define the `postcss-import` options, we add a custom path to our import files using [addModulesDirectories](https://github.com/postcss/postcss-import#addmodulesdirectories)

This allows us to import the CSS file in the Vue `<style>` block without using a relative path, such as `@import 'example';`

I've tried setting this up on Nuxt 3, but to no avail. Here is the configuration:

```javascript
export default defineNuxtConfig({
  postcss: {
    plugins: {
      "postcss-import": {
        addModulesDirectories: ["assets/css/imports"]
      }
    }
  }
});
```

The `resolve` function also doesn't seem to work. I even modified the `node_modules/@nuxt/schema/dist/index.mjs` file directly and added the path, but still no luck. Lastly, I added the path to the `modulesDir` [Nuxt 3 config](https://nuxt.com/docs/api/configuration/nuxt-config#modulesdir). Nothing seems to work. 

Relative paths for the imports still work as expected. [Here](https://codesandbox.io/s/broken-postcss-import-plugin-with-nuxt-3-v2-ghw48t?file=/nuxt.config.ts:63-353) is a codesandbox link showing the error in action. 

Any thoughts?

**Comment by Baroshem** [comment]
Hey,

not sure if I understand correctly but what about using global style imports for Vite?

https://nuxt.com/docs/getting-started/assets#global-styles-imports

**Reply by scottcarlson** [reply]
That appears to only be for CSS preprocessors. It did lead me to try with the Vite [css.postcss](https://vitejs.dev/config/shared-options.html#css-postcss) configurations directly, but it still didn't to work and threw the same error:

```javascript
vite: {
    css: {
        postcss: {
            plugins: [
                postcssImport({
                    resolve() {
                        return ['assets/css/imports'];
                    },
                }),
            ],
        },
    },
},
```

We could of course move to another solution with preprocessors, but the idea is to make the port from Nuxt 2 to Nuxt 3 as seamless as possible, and we'd like to avoid updating all of our `@imports` to relative paths. We may not have a choice at this time, but I'm still unclear if this is a bug or if we're still missing some different implementation strategy/syntax.

To clarify, setting `addModulesDirectories` works as expected in Nuxt 2, and in Nuxt 2 we're able to use PostCSS imports using global module file names instead of relative paths. So if we had some mixins defined in a file at `<rootDir>/assets/css/imports/icons.css`, we could simply add `@import 'icons';` at the top of out `<style>` section in a SFC and it would import the PostCSS mixins as expected. 

So far we can't get this to work in Nuxt 3, so we'd have to update several hundred components to use relative paths, such as `@import './../../assets/css/imports/icons';` Besides the tedious nature of the task, this is less scalable for being able to refactor our application in the future if directory structures change.
